lib/usecases/run/GetAllRunsUseCase.js

```diff
@@ -310,6 +313,35 @@ class GetAllRunsUseCase {
                 filteringQueryBuilder.where('runNumber').oneOf(...runNumbers);
             }
 
+            if (gaq) {
+                if ((dataPassIds ?? []).length !== 1) {
+                    throw new BadParameterError('Filtering by GAQ is enabled only when filtering with one dataPassId');
+                }
+                const { mcReproducibleAsNotBad = false } = gaq;
+                const [dataPassId] = dataPassIds;
+                const gaqSummary = await qcFlagService.getGaqSummary(dataPassId, { mcReproducibleAsNotBad });
```

Comment:
This seems highly unoptimized? Shouldn't we create QC flags summary view at some point?
If you have a lot of data pass, this would be better filtered out in SQL

Reply:
yes, but here, to filter by GAQ you need to provide one and only one dataPassId